### PR TITLE
Enable SRIOV

### LINKED ISSUE #112: Allow SR-IOV during import
Support for SR-IOV will be available within the same release as `vm-import-operator` and we should allow it.

### PR DESCRIPTION
**Pull request by pkliczewski**
In this PR we add ability to import virtual machines connected to SRIOV network.

Fixes:  https://github.com/kubevirt/vm-import-operator/issues/112

Signed-off-by: Piotr Kliczewski <piotr.kliczewski@gmail.com>

### CODE REVIEW

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go [hunk 1]
```diff
@@ -544,6 +544,9 @@ func (r *ReconcileVirtualMachineImport) createVM(provider provider.Provider, ins
 
 	// Create kubevirt VM from source VM:
 	reqLogger.Info("Creating a new VM", "VM.Namespace", vmSpec.Namespace, "VM.Name", vmSpec.Name)
+
+	vmJSON, _ := json.Marshal(vmSpec)
+	reqLogger.Info("VM struct", "VM spec", string(vmJSON))
```
Comment (jakub-dzon) [inline pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: It can be a lot of text. Should it be here?

**Comment by pkliczewski** [comment]
@phoracek please review

File: pkg/providers/ovirt/mapper/mapper.go [hunk 2]
```diff
@@ -339,6 +347,16 @@ func (o *OvirtMapper) mapNics(networkToType map[string]string) []kubevirtv1.Inte
 	return kubevirtNics
 }
 
+func isSRIOV(vNicProfile *ovirtsdk.VnicProfile) bool {
+	sriov := false
```
Comment (jakub-dzon) [inline pkg/providers/ovirt/mapper/mapper.go]: WDYT about resigning from using `sriov` local variable?

**Review by jakub-dzon — COMMENTED** [review]
Would it be possible to add a functional test that imports SRIOV-enabled VM but doesn't start it in KubeVirt? We would at least confirm that the target VM structure is correct.

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go (hunk 1 above)
Comment (pkliczewski) [reply pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: There is a bit of text but I found it useful more than once.

**Comment by pkliczewski** [comment]
I still need to tests edge cases and will explore whether it would be possible to add functional test to cover this feature

File: pkg/providers/ovirt/mapper/mapper.go [hunk 3]
```diff
@@ -339,6 +347,16 @@ func (o *OvirtMapper) mapNics(networkToType map[string]string) []kubevirtv1.Inte
...
+	if pt, ok := vNicProfile.PassThrough(); ok {
+		if ptm, ok := pt.Mode(); ok && string(ptm) == "enabled" {
```
Comment (machacekondra) [inline pkg/providers/ovirt/mapper/mapper.go]: Wouldn't it be better to use:

```golang
if ptm, ok := pt.Mode(); ok && *ptm == ovirtsdk.VNICPASSTHROUGHMODE_ENABLED
```

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go (hunk 1 above)
Comment (masayag) [reply pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: I'm in favor of logging for debug / error handling, especially when we failed to create the VM - we won't be able to debug the VM request easily.

However, how about moving this inside the condition - since if we managed to create the VM, we'll have the VM resource created on etcd to which we can refer for debug.
The main value as I see it if we failed to create the VM.

Also, perhaps yaml format ?

Comment (pkliczewski) [reply pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: Good idea, will move. Json takes one line - no spaces or formatting.

**Comment by phoracek** [comment]
s/SROIV/SR-IOV/

File: go.sum [hunk 4]
```diff
@@ -281,6 +283,8 @@ github.com/fatih/structtag v1.1.0 h1:6j4mUV/ES2duvnAzKMFkN6/A5mCaNYPD3xfbAkLLOF8
 github.com/fatih/structtag v1.1.0/go.mod h1:mBJUNpUnHmRKrKlQQlmCrh5PuhftFbNv8Ys4/aAZl94=
 github.com/flynn/go-shlex v0.0.0-20150515145356-3f9db97f8568/go.mod h1:xEzjJPgXI435gkrCt3MPfRiAkVrwSbHsst4LCFVfpJc=
 github.com/fortytw2/leaktest v1.3.0/go.mod h1:jDsjWgpAGjm2CA7WthBh/CdZYEPF31XHquHwclZch5g=
+github.com/fromanirh/pack8s v0.1.0 h1:n7okFfusL81MTmtlVhWFZcss8JtMxi1HeOwUe6SkJkk=
```
Comment (phoracek) [inline go.sum:286]: Is this the same as the pack8s kept under kubevirt/kubevirtci https://github.com/kubevirt/kubevirtci/tree/master/pack8s?

File: pkg/providers/ovirt/mapper/mapper.go [hunk 5]
```diff
@@ -375,26 +383,37 @@ func (o *OvirtMapper) getNetworkForNic(vnicProfile *ovirtsdk.VnicProfile) kubevi
 	network, _ := vnicProfile.Network()
 	nicNetworkName, _ := network.Name()
 	vnicProfileName, _ := vnicProfile.Name()
+	sroiv := outils.IsSRIOV(vnicProfile)
```
Comment (phoracek) [inline pkg/providers/ovirt/mapper/mapper.go]: sriov

File: pkg/providers/ovirt/mapper/mapper.go [hunk 6]
```diff
@@ -375,26 +383,37 @@ func (o *OvirtMapper) getNetworkForNic(vnicProfile *ovirtsdk.VnicProfile) kubevi
...
-				o.mapNetworkType(mapping, &kubevirtNet)
+				o.mapNetworkType(mapping, &kubevirtNet, sroiv)
 			}
 		}
 	}
 	return kubevirtNet
 }
 
-func (o *OvirtMapper) mapNetworkType(mapping v2vv1.NetworkResourceMappingItem, kubevirtNet *kubevirtv1.Network) {
+func (o *OvirtMapper) mapNetworkType(mapping v2vv1.NetworkResourceMappingItem, kubevirtNet *kubevirtv1.Network, sroiv bool) {
```
Comment (phoracek) [inline pkg/providers/ovirt/mapper/mapper.go]: sriov

File: go.sum (hunk 4 above)
Comment (pkliczewski) [reply go.sum:286]: This is indirect dependency. Was updated by go mod tidy.

**Review by masayag — APPROVED** [review]

**Review by machacekondra — APPROVED** [review]
